apollo-router/src/services/layers/apq.rs

```diff
@@ -164,6 +164,12 @@ fn redis_key(query_hash: &str) -> String {
     format!("apq\0{query_hash}")
 }
 
+pub(crate) fn calculate_hash_for_query(query: String) -> String {
```

Comment:
```suggestion
pub(crate) fn calculate_hash_for_query(query: &str) -> String {
```

This does not need to own the query